test/unit/metadata.js

```diff
@@ -772,4 +774,67 @@ describe('metadata', function () {
       should(upToDate(doc).errors).be.undefined()
     })
   })
+
+  describe('generated a doc with conflict', () => {
+    it('should get the correct path', () => {
+      const doc = {
+        path: 'docname'
+      }
+      const newDoc = createConflictingDoc(doc)
+      const pathRegExp = withConflictRegExp(doc.path)
+      should(newDoc.path).match(pathRegExp)
+    })
+    it('should get the correct _id', () => {
+      const doc = {
+        path: 'docname'
+      }
+      const newDoc = createConflictingDoc(doc)
+      const pathRegExp = withConflictRegExp(doc.path)
+      should(newDoc._id).match(pathRegExp)
+    })
+    it('should keep the correct extension', () => {
+      const ext = '.pdf'
+      const doc = {
+        path: `docname${ext}`
+      }
+      const newDoc = createConflictingDoc(doc)
+      should(path.extname(newDoc.path)).equal(ext)
+    })
+    it('should keep the correct extension when extension is complex', () => {
+      const ext = '.tar.gz'
+      const doc = {
+        path: `docname${ext}`
+      }
+      const newDoc = createConflictingDoc(doc)
+      // FIXME: must be docname-conflict-:ISODATE:.tar.gz instead of docname.tar-conflict-:ISODATE:.gz
```

Comment:
Maybe we could describe the current behavior in the test description, so one doesn't believe it actually works looking at the test results?

Reply:
Yes, you are right, my mistake.